File: easy/2574-left-and-right-sum-differences/solution.py
from typing import List


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        """
        2574. Left and Right Sum Differences
        Time: O(n)
        Space: O(1)
        """
        total = sum(nums)
        left_sum = 0
        answer = []

        for num in nums:
            right_sum = total - left_sum - num
            answer.append(abs(left_sum - right_sum))
            left_sum += num

        return answer

File: easy/2574-left-and-right-sum-differences/solution.java
/**
 * 2574. Left and Right Sum Differences
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int total = 0;
        for (int num : nums) {
            total += num;
        }

        int leftSum = 0;
        int[] answer = new int[nums.length];

        for (int index = 0; index < nums.length; index++) {
            int rightSum = total - leftSum - nums[index];
            answer[index] = Math.abs(leftSum - rightSum);
            leftSum += nums[index];
        }

        return answer;
    }
}
